src/protocols/transaction.rs

```diff
@@ -60,13 +60,11 @@ impl Protocol for TxSubmissionProtocol {
     fn get_agency(&self) -> Agency {
         return match self.state {
             State::Idle => { Agency::Server }
-            State::TxIdsBlocking => {
-                // Typically, this would be client agency, but we pretend it's none since
-                // we just hang on to Client agency and never send any transactions.
-                Agency::None
-            }
+            State::TxIdsBlocking => { Agency::Client }
             State::TxIdsNonBlocking => { Agency::Client }
-            State::Done => { Agency::None }
+            State::Done => {
+                if self.result.is_none() { Agency::Client } else { Agency::None }
```

Comment:
It looks like there should be two different states, one mapped to Agency::Client and the other to Agency::None, to keep a consistent state machine.

Reply:
We move to `State::Done` before we've set the result of the protocol. This lets us hang onto `Agency::Client` agency for one more loop so we can set the result and then move to the `Agency::None` state. I did it this way to more closely match the spec for the protocol.  It can be refactored if/when we implement the full spec for this protocol if we want to actually submit transactions.  Right now, this one just runs enough to tell the server we won't be sending them any transactions.

Reply:
Sounds good to me, and I am linking this comment in the respective ticket...